float64 encoder1
float64 encoder2
float64 encoder3
float64 encoder4

bool direction1
bool direction2
bool direction3
bool direction4
